sidechain/contracts/group/Group.sol

```diff
@@ -309,31 +488,233 @@ contract Group is GroupI, UsingExternalStorage
+                    getSigner(
+                        prepareRevokeInvitation(
+                            _groupId, 
+                            _secretHash, 
+                            _nonce
+                        ), 
+                        _signature, 
+                        _nonce
+                    )
+                )
+            )
+        ); 
+
+        require(uint8(signerRole) == admin);
+
+        storageContract.putUintValue(keccak256(
+            abi.encodePacked(INVITATION_KEY, _groupId, _secretHash, "STATE")),
+            uint(InvitationState.Revoked)
+        );
+
+        return true;
+    }
+
+    /*
+     *  Accepting an Invitation
+     */ 
+
+    function prepareAcceptInvitationCommit(
+        uint256 _groupId,
+        bytes32 _addressSecretHash,
+        uint256 _nonce
+    )
+        public
+        pure 
+        returns (bytes32)
+    {
+        return keccak256(
+            abi.encodePacked(
+                _groupId,
+                "acceptInvitationCommit",
+                _addressSecretHash,
+                _nonce
+            )
+        );
+    }
+
+    function acceptInvitationCommit(
+        uint256         _groupId,
+        bytes32         _addressSecretHash,
+        bytes memory    _signature,
+        uint256         _nonce
+    )
+        public
+        returns (bool)
+    {
+        // get sender by retrieving from signature
+        address sender = getSigner(
+            prepareAcceptInvitationCommit(
+                _groupId,
+                _addressSecretHash,
+                _nonce
+            ), 
+            _signature, 
+            _nonce
+        );
+
+        commits[sender].id          = _groupId;
+        commits[sender].commit      = _addressSecretHash;
+        commits[sender].sig         = _signature;
+        commits[sender].block       = uint64(block.number);
+        commits[sender].revealed    = false;
+
+        emit AcceptCommitted(_groupId, _addressSecretHash);
+        return true;
+    }
+
+    function acceptInvitation(
+        uint256 _groupId,
+        bytes32 _secret,
+        address _sender
+    )
+        public
+        returns (bool)
+    {
+        // assign reference to ephemeral Commit struct
+        Commit storage tempCommit = commits[_sender];
+        
+        // retrieve secretHash
+        bytes32 secretHash      = storageContract.getBytes32Value(keccak256(
+            abi.encodePacked(INVITATION_KEY, _groupId, keccak256(abi.encodePacked(_secret)), "SECRET_HASH")
+        ));
+
+        // retrieve expiration date from ext. storage 
+        uint expirationDate     = storageContract.getUintValue(keccak256(
+            abi.encodePacked(INVITATION_KEY, _groupId, _secret, "EXPIRATION_DATE")
+        ));
+
+        // retrieve preset role for invited user from ext. storage
+        uint role               = storageContract.getUintValue(keccak256(
+            abi.encodePacked(INVITATION_KEY, _groupId, _secret, "ROLE")
```

Comment:
The key to retrieve the role is incorrect so I imagine the storage contract return 0.

In **storeInvitation()**, the key used is `abi.encodePacked(INVITATION_KEY, _groupId, _secretHash, "ROLE"))`

But here it's `abi.encodePacked(INVITATION_KEY, _groupId, _secret, "ROLE")`

Need to replace `_secret` by `keccak256(abi.encodePacked(_secret))`